### PR TITLE
chore(telemetry) improve github action and circle detection

### PR DESCRIPTION
**Pull request by jamo**
Improves `ENV` based CI detection. This would be cool to backport

### CODE REVIEW

File: packages/gatsby-core-utils/src/ci.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-  envFromCIAndCIName,
   getEnvDetect({ key: `NOW_BUILDER_ANNOTATE`, name: `ZEIT Now` }),
   getEnvDetect({ key: `NOW_REGION`, name: `ZEIT Now v1` }),
   getEnvDetect({ key: `VERCEL_URL`, name: `Vercel Now` }),
   getEnvDetect({ key: `NOW_BUILDER`, name: `Vercel Now` }),
   getEnvDetect({ key: `VERCEL_BUILDER`, name: `Vercel Now` }),
   getEnvDetect({ key: `CODESANDBOX_SSE`, name: `CodeSandbox` }),
+  getEnvDetect({ key: `GITHUB_ACTIONS`, name: `GitHub Actions` }),
+  getEnvDetect({ key: `CIRCLE_BRANCH`, name: `CircleCI` }),
+  getEnvDetect({ key: `CIRCLECI`, name: `CircleCI` }),
```
Comment (pieh) [inline packages/gatsby-core-utils/src/ci.ts:12]: Just for papertrail - CircleCi env vars are document in https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables (at least as of moment of writing this comment heh)

File: packages/gatsby-core-utils/src/ci.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 
 const CI_DEFINITIONS = [
-  envFromCIAndCIName,
   getEnvDetect({ key: `NOW_BUILDER_ANNOTATE`, name: `ZEIT Now` }),
   getEnvDetect({ key: `NOW_REGION`, name: `ZEIT Now v1` }),
   getEnvDetect({ key: `VERCEL_URL`, name: `Vercel Now` }),
   getEnvDetect({ key: `NOW_BUILDER`, name: `Vercel Now` }),
   getEnvDetect({ key: `VERCEL_BUILDER`, name: `Vercel Now` }),
   getEnvDetect({ key: `CODESANDBOX_SSE`, name: `CodeSandbox` }),
+  getEnvDetect({ key: `GITHUB_ACTIONS`, name: `GitHub Actions` }),
```
Comment (pieh) [inline packages/gatsby-core-utils/src/ci.ts:10]: Documented in https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables (at least as of moment of writing this comment)

**Review by pieh — APPROVED** [review]
Thanks!

**Comment by vladar** [comment]
Published in `gatsby-core-utils@1.7.1` and `gatsby@2.29.2`

**Pull request merged**